pkg/controller/servicebindingrequest/nested/nested_test.go

```diff
@@ -0,0 +1,276 @@
+package nested
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetValue(t *testing.T) {
+	type args struct {
+		src      map[string]interface{}
+		path     string
+		expected interface{}
+	}
+
+	assertGetValue := func(args args) func(t *testing.T) {
+		return func(t *testing.T) {
+			actual, found, err := GetValue(args.src, args.path, args.path)
+			require.NoError(t, err)
+			require.True(t, found)
+			require.Equal(t, args.expected, actual)
+		}
+	}
+
+	t.Run("key", assertGetValue(args{
+		src: map[string]interface{}{
+			"key": "value",
+		},
+		path: "key",
+		expected: map[string]interface{}{
+			"key": "value",
+		},
+	}))
+
+	t.Run("key.subKey0", assertGetValue(args{
+		src: map[string]interface{}{
+			"key": map[string]interface{}{
+				"subKey0": "value0",
+				"subKey1": "value1",
+			},
+		},
+		path: "key.subKey0",
+		expected: map[string]interface{}{
+			"key": map[string]interface{}{
+				"subKey0": "value0",
+			},
+		},
+	}))
+
+	t.Run("key.slice", assertGetValue(args{
+		src: map[string]interface{}{
+			"key": map[string]interface{}{
+				"subKey0": "value0-0",
+				"slice": []map[string]interface{}{
+					{"subKey0": "value0-1"},
+				},
+			},
+		},
+		path: "key.slice",
+		expected: map[string]interface{}{
+			"key": map[string]interface{}{
+				"slice": []map[string]interface{}{
+					{"subKey0": "value0-1"},
+				},
+			},
+		},
+	}))
+
+	t.Run("key.slice.0", assertGetValue(args{
+		src: map[string]interface{}{
+			"key": map[string]interface{}{
+				"slice": []map[string]interface{}{
+					{"subKey0": "value0"},
+					{"subKey1": "value1"},
+				},
+			},
+		},
+		path: "key.slice.0",
+		expected: map[string]interface{}{
+			"key": map[string]interface{}{
+				"slice": map[string]interface{}{
+					"subKey0": "value0",
+				},
+			},
+		},
+	}))
+
+	t.Run("key.slice.1", assertGetValue(args{
+		src: map[string]interface{}{
+			"key": map[string]interface{}{
+				"slice": []map[string]interface{}{
+					{"subKey0": "value0"},
+					{"subKey1": "value1"},
+				},
+			},
+		},
+		path: "key.slice.1",
+		expected: map[string]interface{}{
+			"key": map[string]interface{}{
+				"slice": map[string]interface{}{
+					"subKey1": "value1",
+				},
+			},
+		},
+	}))
+
+	t.Run("key.slice.*", assertGetValue(args{
```

Comment:
Question -> `key.slice.*` is same as `key.slice`, right?

Reply:
Yes, it is the same if `key.slice` is an actual slice.